### PR TITLE
buildah: increase rootless test coverage with `rootless + unshare`

### PR DESCRIPTION
**Pull request by flouthoc**
Tests for `rootless` added here https://github.com/containers/buildah/pull/3804 skipped some of the cases. 

Almost all of the skipped test cases can be covered under an environment of `rootless + unshare`. 
Following PR attempts to solve that by modifying test infrastructure and adding a `cirrus` task dedicated to `rootless+unshare` testing.

### CODE REVIEW

**Comment by edsantiago** [comment]
I think there's some confusion here. What I meant to do with my patch is _eliminate the need to run the whole entire thing unshared_.

With my patch, you can remove a lot of the `skips`: the tests will pass under plain simple rootless. `unshare` is only used for `buildah mount/unmount`, so it's a better test of actual rootless.

Running the entire test suite under `unshare` does not seem terribly useful to me

**Comment by flouthoc** [comment]
@edsantiago Yes. Actually this is intentional. My motive is to perform isolated testing for  `rootless` and `rootless+unshare` in different environment cause they both are testing different things. 

With this we can make sure that `user namespacing` is working correctly.

**Comment by edsantiago** [comment]
I have no real idea what that means, but given what I see, you do not need my patch for anything: my patch was intended to make more tests pass. You have not removed any skips, you have only added more skips. That is the opposite of my intention.

Please either remove skips, or remove my patch.

**Comment by flouthoc** [comment]
@edsantiago  Maybe my response above was a bit confusing but I'll try to re-pharse it and please correct me if I got it wrong anywhere.

So motivation behind this is that `rootless buildah` and `rootless buildah unshare` does fundamentally different things.

While `rootless buildah` **does not** creates its own `user namespace` hence it is has limited functionality but it still supports most of the feature which is tested here: https://github.com/containers/buildah/pull/3804

OTOH `rootless buildah unshare` creates it own `user namespace` including the `mount namespace` and **almost all** of the features work with `rootless buildah unshare` which works with `buildah root`

Therefore I think they provide different environment, my motivation is just to test these two differently.

**Comment by flouthoc** [comment]
> I have no real idea what that means, but given what I see, you do not need my patch for anything: my patch was intended to make more tests pass. 

@edsantiago I think this we can still do this. Scratch my previous comment and let me try the original patch which you posted again since it only works for `mount / unmount` i think it does what we are trying to achieve.

**Comment by flouthoc** [comment]
@edsantiago Thanks :smile: . I looked at it again and re-used your patch as is it and this enables us to remove skip from almost all of the `rootless` tests some are still skipped but they are more of a `root` user test so we should be good.

@rhatdan @cevich @nalind @edsantiago PTAL

File: tests/helpers.bash [hunk 1]
```diff
@@ -160,6 +160,32 @@ function podman() {
     command podman ${PODMAN_REGISTRY_OPTS} ${ROOTDIR_OPTS} "$@"
 }
 
+# Used in rootless
```
Comment (edsantiago) [inline tests/helpers.bash:163]: A future maintainer might appreciate some thorough comments here

File: tests/helpers.bash [hunk 2]
```diff
@@ -195,6 +221,12 @@ function run_buildah() {
     # Remember command args, for possible use in later diagnostic messages
     MOST_RECENT_BUILDAH_COMMAND="buildah $*"
 
+    if is_rootless; then
```
Comment (edsantiago) [inline tests/helpers.bash:224]: ...and especially here

**Review by edsantiago — COMMENTED** [review]
Thank you @flouthoc, that is exactly what I intended. 

My patch was hasty and did not include comments; could you please add some for benefit of future maintainers?

**Comment by edsantiago** [comment]
Oh, one more. Since there's no longer the need for multiple confusing `rootless_this_and_that`, could you just rename everything to `skip_if_rootless`?

**Comment by flouthoc** [comment]
@edsantiago Thanks added requested comments.

> Oh, one more. Since there's no longer the need for multiple confusing `rootless_this_and_that`, could you just rename everything to `skip_if_rootless`?

I agree I think we can merge all existing function into one. I don't know why did we add `$BUILDAH_ISOLATION" = "rootless"
` check and `skip_if_rootless_and_cgroupv1` we might not even need all of this now. But I just want to go through commit history once before removing these. But since this look unrelated to this PR I'll do that in a new PR after reading commit history.

**Review by cevich — APPROVED** [review]
LGTM

**Comment by cevich** [comment]
Oh, and super-big thanks for doing this, it really helped everyone out a lot, especially me :grin:

**Review by edsantiago — COMMENTED** [review]
Language suggestions. Also, I have a medium-strong preference for eighty-column formatting especially in comments. I don't think we have a style guide, so I will not make my preference a blocker.

I see what you mean re: `skip_if_rootless` but we seem to have a very very bad situation that needs to be fixed: the existing `skip_if_rootless` bears no relation whatsoever to the `is_rootless()` function, and that is going to cause a lot of confusion one day (including today). That would be a very good topic for a followup PR.

**Comment by flouthoc** [comment]
Restarted flakes.

**Pull request merged**